tiff-2-png-converter.py

```diff
@@ -1,197 +1,358 @@
-    if verbose:
-        print("resolution_image_before " + str(width)+ " "+str( height))  
-        print("downscale " + str(downscale))  
-	
-    resolution_image = ( int(width / downscale), int(height / downscale))
-    
-    count = 0
-    image_out = np.zeros(())
-    for i in range(dim):
-        imgs_row = np.zeros(())
-        for j in range(dim):
-            tmp_img = np.zeros((resolution_image))
-            if count < len(volume) and count <= depth:
-               img =volume[count]
-               tmp_img =  cv2.resize(img, resolution_image)
-            else:          
-                img =volume[0]     
-                tmp_img =  cv2.resize(img,resolution_image)
-                tmp_img[:,:] = 0
-            if j == 0:
-                imgs_row = np.copy(tmp_img)
+            print("ERROR: Verify source and destination paths exists")
+            sys.exit()
+
+        if args.verbose:
+            TiffProcessor.verbose = True
+        if args.equalize1 or args.equalize2:
+            TiffProcessor.equalize_histogram = True
+            if args.equalize1:
+                TiffProcessor.equalize_histogram_method = (
+                    equalize_image_histogram_opencv
+                )
+            if args.equalize2:
+                TiffProcessor.equalize_histogram_method = (
+                    equalize_image_histogram_custom
+                )
+
+        if args.gamma:
+            TiffProcessor.gamma = True
+            TiffProcessor.gamma_val = args.gamma
+        if args.num_imgs:
+            TiffProcessor.num_imgs = int(args.num_imgs)
+
+    def convert_to_png(self):
+        list_files, width, heigth = self.list_folder_files(self.source)
+        if self.num_imgs is None:
+            self.num_images = len(list_files)
+        image_r = [None] * self.num_imgs
+        image_g = [None] * self.num_imgs
+        image_b = [None] * self.num_imgs
+        images = [None] * self.num_imgs
+        bits_per_sample = ""
+        print_progress_bar(
+            0, self.num_imgs, prefix="grouping channels:", suffix="Complete", length=50
+        )
+        for i in range(len(list_files)):
+            file_full_path = list_files[i]
+            img = tiff.imread(list_files[i])
+            ## check all images have the same data type
+            if bits_per_sample == "":
+                bits_per_sample = img.dtype
+            elif bits_per_sample != img.dtype:
+                print(
+                    "image "
+                    + file_full_path
+                    + "has a different sample type "
+                    + bits_per_sample
+                    + " expected: "
+                    + bits_per_sample
+                )
+                sys.exit("Check images bits per sample tag")
+
+            name, extension = os.path.splitext(file_full_path)
+            file_name = os.path.basename(name)
+            sequence_number = re.search("^p[0-9]{1,5}", file_name)
```

Comment:
is this file naming method universal to all tiffs? (I know almost nothing about tiffs)